crypto/include/internal/evp_int.h

```diff
@@ -385,6 +410,9 @@ struct evp_pkey_st {
 # endif
 # ifndef OPENSSL_NO_EC
         struct ec_key_st *ec;   /* ECC */
+        CURVE25519_KEY *curve25519;
+        X448_KEY *x448;
+        ED448_KEY *ed448;
```

Comment:
Please consider making an ECX_KEY where you allocate both members instead.

Reply:
Good idea. I have no idea why I didn't do this to start with. It simplifies the code considerably.